bin/review-rot

```diff
@@ -71,23 +56,134 @@ def main(state_, value, duration, ssl_verify=None):
     sorted_results = sorted(
         results,
         key=operator.attrgetter('time'),
-        reverse=args.reverse,
+        reverse=arguments.get('reverse'),
     )
     for result in sorted_results:
-        print(result.format(style=args.format))
+        print(result.format(style=arguments.get('format', 'oneline')))
+
+
+def load_config_file():
+    """
+       Loads the configuration file from the user's home directory.
+       Returns:
+           config(dict): Returns the configurations
+       """
+    home = expanduser('~')
+    path_to_config = os.path.join(home, '.reviewrot.yaml')
+    if not os.path.exists(path_to_config):
+        raise RuntimeError("No config '.reviewrot.yaml' provided. Please"
+                           " specify it in home directory: %s" % home)
+
+    config = load_ordered_config(path_to_config)
+    if isinstance(config, list):
+        # convert to new format
+        config = dict(git_services=config)
+        answer = str(raw_input("Would you like to rewrite '.reviewrot.yaml"
+                               "' in new format [Y/n] : ")).lower().strip()
+        if answer == 'y':
+            # Take the backup of configuration file and
+            # save the configurations in new format
+            backup_path = path_to_config + '.backup'
+            log.info("Creating back up at " + backup_path)
+            copyfile(path_to_config, backup_path)
+            log.info("Rewriting %r in new format!" % path_to_config)
+            with open(path_to_config, 'w') as f:
+                f.write(yaml.dump(config, default_flow_style=False))
+                f.write("\narguments:\n")
+                f.close()
+
+    return config
+
+
+def load_ordered_config(path_to_config):
+    """
+      Loads the configuration in the same order as it's defined in yaml file.
+      Args:
+            path_to_config (str): Path to configuration file
+      Returns:
+            config(dict): Returns the configurations in the defined ordered
+    """
+    _mapping_tag = yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG
+
+    def dict_representer(dumper, data):
+        return dumper.represent_mapping(_mapping_tag, data.iteritems())
+
+    def dict_constructor(loader, node):
+        return collections.OrderedDict(loader.construct_pairs(node))
+
+    yaml.add_representer(collections.OrderedDict, dict_representer)
+    yaml.add_constructor(_mapping_tag, dict_constructor)
+    # read input from home directory for pull requests
+    with open(path_to_config, 'r') as f:
+        config = yaml.load(f)
+    return config
+
+
+def get_arguments(config_arguments):
+    """
+       Get the arguments provided in configuration file
+       and command line arguments
+       Args:
+             config_arguments (dict): Arguments specified in yaml file
+       Returns:
+             arguments (dict): Returns the parsed arguments
+     """
+    parsed_arguments = {}
+    command_line_args = vars(args)
```

Comment:
Where is `args` defined? It needs to be passed into this function. Some unit tests would be helpful to detect this.